util/settings/settings.go

```diff
@@ -1677,10 +1677,11 @@ func UnmarshalDexConfig(config string) (map[string]interface{}, error) {
 }
 
 func (a *ArgoCDSettings) oidcConfig() *oidcConfig {
-	if a.OIDCConfigRAW == "" {
+	OIDCConfigRAW := ReplaceStringSecret(a.OIDCConfigRAW, a.Secrets)
+	if OIDCConfigRAW == "" {
```

Comment:
@jskrzypek I think after replacement it may be a good idea to update the raw config stored in the `ArgoCDSettings` object as other parts of the code may also reference it
can we add `a.OIDCConfigRAW = OIDCConfigRAW` here?

Reply:
The issue is that if we do that, we lose the original reference to the secret and key that's the source of the config. If the secrets change after that won't we be unable to detect it?

Reply:
Also I looked around and OIDC ConfigRaw seemed to only be consumed here, but if you know of some other place in the coffee where it's used let me know

Reply:
@jskrzypek 
> The issue is that if we do that, we lose the original reference to the secret and key that's the source of the config. If the secrets change after that won't we be unable to detect it?

Yeah sorry my bad, you're right we shouldn't overwrite it in the settings 

> Also I looked around and a.OIDCConfigRaw seemed to only be consumed here, but if you know of some other place in the code where it's used let me know

Yeah there's a few other places that this field is accessed in. They won't show up if you search for `a.OIDCConfigRAW` because `a` is a local variable. But if you search for just `OIDCConfigRAW` they'll show up. It is referenced as `settings.OIDCConfigRAW` or `argosettings.OIDCConfigRAW` in other places. We may want to take a look at those lines and also see if this resolution needs to be added in those places as well

Reply:
I was using vs code's Peek References functionality actually, but maybe it
missed some of the refs that live outside the module... I'll double check
this and ensure that I handle any additional replacements as needed